osquery/tables/networking/posix/curl.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/network/include/http/client.hpp>
+#include <boost/numeric/conversion/cast.hpp>
+#include <chrono>
+#include <osquery/tables.h>
+
+#include <stdio.h>
+
+using namespace boost::network::http;
+using namespace std::chrono;
+
+namespace osquery {
+namespace tables {
+
+void processRequest(const std::string& request_str, QueryData& results) {
```

Comment:
General convention in osquery would be to return a Status from this function. Default Status() constructor is "0, OK", and then in your catch block below you could return Status(1, e.what()).

Then in your calling function, if !return_val.ok(), then LOG(WARN) << return_val.getMessage()